TheMovieDB/Data/MovieDataSource.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  MovieService.swift
+//  TheMovieDB
+//
+//  Created by Intern on 14/07/23.
+//
+
+import Foundation
+
+protocol MovieDataSourceProtocol {
+    func getPopularMovies(page: Int, completion: @escaping (QueryResult?) -> Void)
+}
+
+let AUTH: String = "eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiJmNjNiMTg2NzRmZjlmNzAzYjVlYjU3NjcyYmFmZDljMiIsInN1YiI6IjY0OGFhOTdiMjYzNDYyMDBjYTE4Nzk4YSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.tQgyjS495o4qqwycv2DYovgi544ubIwy6ZtujaiqWok"
+class MovieDataSource: MovieDataSourceProtocol {
+    
+    //@Published var pageResults: QueryResult = QueryResult(page: 0, results: [])
+    func getPopularMovies(page: Int, completion: @escaping (QueryResult?) -> Void) {
+        guard let url = URL(string: "https://api.themoviedb.org/3/movie/popular?language=en-US&page=\(page)") else {
```

Comment:
you can improve your hardcode url using this

```swift
enum SimpleURL {
    case home
    case about
    case products(page: Int)
    
    var baseURL: String {
        return "https://www.example.com"
    }
    
    var path: String {
        switch self {
        case .home:
            return "/"
        case .about:
            return "/about"
        case .products:
            return "/products"
        }
    }
    
    var queryItems: [URLQueryItem]? {
        switch self {
        case .products(let page):
            return [URLQueryItem(name: "page", value: "\(page)")]
        default:
            return nil
        }
    }
    
    var url: URL? {
        var components = URLComponents(string: baseURL)
        components?.path = path
        components?.queryItems = queryItems
        return components?.url
    }
}
```

In this enum, we define three cases: `home`, `about`, and `products`. The `products` case has an associated value of `page` which represents the page argument. 

The enum also provides computed properties such as `baseURL` (the base URL for all cases), `path` (the path specific to each case), `queryItems` (query items specific to each case), and `url` (the final constructed URL).

You can use this enum as follows:

```swift
// Example usage
let homeURL = SimpleURL.home.url // https://www.example.com/
let aboutURL = SimpleURL.about.url // https://www.example.com/about
let productsURL = SimpleURL.products(page: 1).url // https://www.example.com/products?page=1
```

By providing different associated values for the `products` case, you can construct URLs with different page arguments dynamically.